scripts/irods/test/resource_suite.py

```diff
@@ -1167,3 +1167,181 @@ def test_itrim_no_op(self):
 
         # try to trim down to repl_count
         self.admin.assert_icommand("itrim -N {repl_count} {filename}".format(**locals()), 'STDOUT_SINGLELINE', "Total size trimmed = 0.000 MB. Number of files trimmed = 0.")
+
+
+    ###################
+    # detached mode
+    ###################
+    def test_detached_mode__issue_4421(self):
+        try:
+            file1 = "f1"
+            file2 = "f2"
+            resource_host = "irods.org"
+            resource_name = 'detached_resource'
+            resource_context = 'host_mode=detached'
+
+            lib.create_ufs_resource(resource_name, self.admin, resource_host)
+            self.admin.assert_icommand("iadmin modresc %s context %s" %
+                                   (resource_name, resource_context))
+
+            # create file to put
+            lib.make_file(file1, 100)
+
+            # put small file
+            self.admin.assert_icommand("iput -R %s %s" % (resource_name, file1))  # iput
+
+            # get file
+            self.admin.assert_icommand("iget %s %s" % (file1, file2))  # iget
+
+            # make sure the file that was put and got are the same
+            self.admin.assert_icommand("diff %s %s " % (file1, file2))
+        finally:
+            # local cleanup
+            self.admin.assert_icommand("irm -f " + file1)
+
+            if os.path.exists(file1):
+                os.unlink(file1)
+            if os.path.exists(file2):
+                os.unlink(file2)
+
+            # cleanup
+            lib.remove_resource(resource_name, self.admin)
+
+    def test_detached_mode_vault_path_updated__issue_4421(self):
+        try:
+            file1 = "f1"
+            file2 = "f2"
+            resource_host = "irods.org"
+            resource_name = 'detached_resource'
+            vault_path = '/tmp/detached_resource'
+            hostname = lib.get_hostname()
+            resource_context = 'host_mode=detached;host_list=irods.org:/some/path,%s:%s;some_other_host:/some/other/path' % (hostname, vault_path)
```

Comment:
Should the semicolon right before `some_other_host` be a comma?

Reply:
No, the entire string is part of the host_list KVP.  The comma separates host/paths which is not the same as semicolon which separates different KVP's.

Reply:
mmm, i think alan's right?

Reply:
It should be a comma.  Will change.

Reply:
done